sdk/storage/azure-storage-files-shares/src/share_file_client.cpp

```diff
@@ -840,6 +846,11 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
             chunkOptions.Range.Value().Offset = offset;
             chunkOptions.Range.Value().Length = length;
             auto chunk = Download(chunkOptions, context);
+            if (chunk.Value.Details.ETag != etag)
+            {
+              throw Azure::Core::RequestFailedException(
+                  "file was modified in the middle of download");
```

Comment:
```suggestion
                  "File was modified in the middle of download.");
```

Reply:
https://github.com/Azure/azure-sdk-for-cpp/issues/2366